lms/djangoapps/courseware/views/views.py

```diff
@@ -674,6 +674,27 @@ def course_about(request, course_id):
         return render_to_response('courseware/course_about.html', context)
 
 
+@ensure_csrf_cookie
+@cache_if_anonymous()
+def program_detail(request, marketing_slug):
+    """
+    Display the program marketing page.
+    """
+    program = get_program_with_type_and_instructors(marketing_slug)
+
+    if not program:
+        raise Http404
+
+    # TODO: put these keys dynamically
+    for course in program['courses']:
+        course_runs = course['course_runs'][0]
+        course_runs['registered'] = False
+        course_runs['can_enroll'] = True
+        course_runs['is_shib_course'] = False
+
+    return render_to_response('courseware/program_detail.html', {'program': program})
```

Comment:
Instead of passing the full program in the context, we should construct a context object that contains all the values needed in the template. The context object should provide empty values for keys which may not exit in the program dict. This way the page won't blow up if the program is missing a value.